### DISCUSSION TITLE
Updated device tree request for the Xiaomi MI 9T

### DISCUSSION THREAD

**Discussion by Diniboy1123** [Divested-Mobile/DivestOS-Build#121] [Ideas]
Hi,

We have been eagerly waiting for the official LineageOS maintainer to update his device tree to support the latest 19.1 version but this has never happened. There is however another tree which works stable, has all the device features fully implemented and the developer himself is a well known part of the official Lineage community and he pushes numerous gerrit changes plus maintains other devices officially for lineage. He also has an unofficial, but extremely stable 19.1 build for the MI 9T too.

Here are the sources:

https://github.com/xiaomi-sm6150/android_device_xiaomi_davinci
https://github.com/xiaomi-sm6150/android_device_xiaomi_sm6150-common
https://github.com/xiaomi-sm6150/android_vendor_xiaomi

Would you consider switching to this one please? I believe it's better for security as well to support something that's not EOL by nature from LineageOS either.

**Comment by reyis4444** [comment]
@SkewedZeppelin 
It would be really nice to see that.

https://www.reddit.com/r/LineageOS/comments/v7wrjc/a_complaint_about_mi_9t_and_the_developer/

**Comment by rsssxd** [comment]
Is it possible to get 19.1 update for Davinci soon?
![IMG_20220619_211955_107](https://user-images.githubusercontent.com/60405617/174495023-67752a5a-9ed7-4579-b4bc-e9dba55dcf46.jpg)

**Comment by rsssxd** [comment]
https://github.com/sm6150-dev/android_kernel_xiaomi_sm6150

**Reply by rsssxd** [reply]
@SkewedZeppelin 

Build maintained by official developer

**Comment by rsssxd** [comment]
https://github.com/adi8900

**Comment by GarciaLnk** [comment]
I've compiled DivestOS 19.1 for davinci using the device tree from xiaomi-sm6150 and I can confirm that it works flawlessly. I don't know why it hasn't gotten official status in LineageOS, assuming the maintainers have bothered in submitting the port.

**Reply by rsssxd** [reply]
Do you have any Telegram address? I would like to test and use the build you compiled.

**Reply by rsssxd** [reply]
@SkewedZeppelin 

There is a build that has been verified to work and was developed by the previous official developer.

Is it possible to switch?

**Reply by SkewedZeppelin** [reply]
@GarciaLnk
Awesome, I'll give it a try sometime soonish, perhaps after the July build cycle.

My biggest issue is with the handling of blobs and their compatibility with the other existing Xiaomi devices.

**Comment by rsssxd** [comment]
IBuilding on the 19.1 build developed by Pig, I was able to run DivestOS without any problems.

**Comment by rsssxd** [comment]
@SkewedZeppelin 
No A12 update for Davinci this month. A new building was to be build after July.

**Comment by rsssxd** [comment]
@SkewedZeppelin Why is Davinci not receiving updates?

**Reply by SkewedZeppelin** [reply]
LineageOS removed some files due to legal concerns, it doesn't compile anymore.

**Reply by rsssxd** [reply]
There are many builds available for the A12. It can be rebuilt on them.

There are many users who can test it and provide feedback.

**Comment by rsssxd** [comment]
![Screenshot_20221122-211123_Trebuchet](https://user-images.githubusercontent.com/60405617/203521167-6acf68e4-43bd-482f-8f94-c9a2d952f455.png)
![Screenshot_20221122-211127_Trebuchet](https://user-images.githubusercontent.com/60405617/203521191-a359aebe-1a82-44d7-be8b-3644bcc7f936.png)
![Screenshot_20221122-211136_Settings](https://user-images.githubusercontent.com/60405617/203521193-0df2d419-e25e-48d2-ab6f-afcc47c41747.png)
![Screenshot_20221122-211145_Settings](https://user-images.githubusercontent.com/60405617/203521196-12775778-3db8-427d-a343-a761a9af21ef.png)
![Screenshot_20221122-211201_Settings](https://user-images.githubusercontent.com/60405617/203521201-16e21aef-9265-4d99-9dad-82e435f14a3c.png)
![Screenshot_20221122-211225_Settings](https://user-images.githubusercontent.com/60405617/203521205-4b0fa4b7-e2a4-488e-934e-a6946e2081b8.png)

**Comment by rsssxd** [comment]
Hello

@SkewedZeppelin 

There is a kernel structure for Davinci that has CFI and SCS protections enabled, and has been successfully tested by many users.

Is it possible to replace this with the current kernel for davinci?

https://github.com/DavinciCodeOS/kernel_xiaomi_sm6150

**Comment by rsssxd** [comment]
@SkewedZeppelin Davinci has an old firmware update. Is this so hard to fix?

**Comment by rsssxd** [comment]
https://download.lineageos.org/devices/davinci/builds

**Reply by SkewedZeppelin** [reply]
yes, it is staged here: https://github.com/Divested-Mobile/DivestOS-Build/pull/234
I'll be getting to it soon
